release/models/qos/openconfig-qos-mem-mgmt.yang

```diff
@@ -271,18 +277,40 @@ submodule openconfig-qos-mem-mgmt {
       type uint64;
       units bytes;
       description
-        "The mininum threshold parameter for a RED-managed queue. When the
-        average queue length is less than minth, all packets are admitted
-        to the queue.";
+        "The mininum threshold parameter for a RED-managed queue in bytes.
+        When the average queue length is less than minth, all packets are admitted
+        to the queue.  Mututally exclusive with min-threshold-percent and
+        max-threshold-percent.";
     }
 
     leaf max-threshold {
       type uint64;
       units bytes;
       description
-        "The maximum threshold parameter for a RED-managed queue. When the
-        average queue length exceeds the maxth value, all packets are
-        dropped (or marked if ECN is enabled).";
+        "The maximum threshold parameter for a RED-managed queue in bytes.
+        When the average queue length exceeds the maxth value, all packets are
+        dropped (or marked if ECN is enabled).  Mutually exclusive with
+        min-threshold-percent and max-threshold-percent.";
+    }
+
+    leaf min-threshold-percent {
+      type uint64;
+      units oc-types:percentage;
+      description
+        "The mininum threshold parameter for a RED-managed queue in percent.
+        When the average queue length is less than minth, all packets are
+        admitted to the queue.  Mutually exclusive with min-threshold and
+        max-threshold.";
+    }
+
+    leaf max-threshold-percent {
+      type uint64;
+      units oc-types:percentage;
+      description
+        "The maximum threshold parameter for a RED-managed queue in percent.
+        When the average queue length exceeds the maxth value, all packets
+        are dropped (or marked if ECN is enabled).  Mutually exclusive with
+        min-threshold and max-threshold.";
     }
 
     leaf enable-ecn {
```

Comment:
Not directly related to this change, but we recently noticed issue #861, and it may be worth trying to fix this at the same time.

Reply:
This Pull is open for a long time. Can we please detach from fresh requests and merge so we can unblock other efforts?

Reply:
Thanks for the comments @rgwilton and @sachendras .   We will merge this first and address #861 separately.